File: Java/Sorting/quick_sort.java
import java.util.Scanner;

class quick_sort{
	//function to swap values in array
	void swap_values(int[] arr, int i, int j){
		int temp = arr[i];
		arr[i] = arr[j];
		arr[j] = temp;
	}
	//puts all the values<pivot to left of pivot
	//and greater to the left
	int partition(int[] arr, int low, int high){
		int pivot = arr[high];
		int i = (low - 1);
		for (int j = low; j <= high - 1; j++) {
			if (arr[j] < pivot) {
				i++;
				swap_values(arr, i, j);
			}
		}
		swap_values(arr, i + 1, high);
		return (i + 1);
	}
    //function for quick sort
	void quickSort(int[] arr, int low, int high){
		if (low < high) {
			int pivot = partition(arr, low, high);
			
			quickSort(arr, low, pivot - 1);
			quickSort(arr, pivot + 1, high);
		}
	}

	public static void main(String[] args)
	{
	    Scanner s = new Scanner(System.in);    
	    //enter array size
        int n = s.nextInt();
        //initialize array with size n
        int[] arr = new int[n];
        //input the array
        for(int i=0;i<n;i++){
            arr[i] = s.nextInt();   
        }
		s.close();

		quick_sort sortArray = new quick_sort();
        //call the quickSort function
		sortArray.quickSort(arr, 0, n - 1);
		//print the array after sorting
		System.out.println("Sorted array: ");
		for (int i = 0; i < n; i++){
			System.out.print(arr[i] + " ");
		}
	}
}

File: C++/Sorting/quick_sort.cpp
#include<bits/stdc++.h>

using namespace std;
int hPartition(int a[], int low, int high)
{
    int pivot = a[low];
    int i = low - 1;
    int j = high + 1;
    while (true)
    {
        do
        {
            i++;
        } while (a[i] < pivot);
        do
        {
            j--;
        } while (a[j] > pivot);
       
        if (i >= j)
            return j;
            
             swap(a[i], a[j]);
    }
}

void quickSort(int a[],int l,int h){
    if(l<h){
        int p = hPartition(a,l,h);
        quickSort(a,l,p);
        quickSort(a,p+1,h);
    }
}

int main()
{
int a[] = {5,2,1,100,22,33,-1};
int n = sizeof(a)/sizeof(int);
quickSort(a,0,n-1);
for(auto i : a)
cout<<i<<" ";
return 0;
}